src/rpc/wallet_api.rs

```diff
@@ -222,3 +222,36 @@ where
     let ret = sig.verify(&msg, &address).is_ok();
     Ok(ret)
 }
+
+/// Deletes a wallet given its address.
+pub(in crate::rpc) async fn wallet_delete<DB>(
+    data: Data<RPCState<DB>>,
+    Params(params): Params<WalletDeleteParams>,
+) -> Result<WalletDeleteResult, JsonRpcError>
+where
+    DB: Blockstore,
+{
+    let (addr_str,) = params;
+    let mut keystore = data.keystore.write().await;
+    let addr = Address::from_str(&addr_str)?;
+    crate::key_management::remove_key(&addr, &mut keystore)?;
+
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::{shim::crypto::SignatureType, KeyStore};
+
+    #[tokio::test]
```

Comment:
Let's make them more exhaustive. I'd like to see an attempt to delete a non-existing key (both because the keystore is empty and because it was not inserted) and some logic to validate what happens to the default key after removal.